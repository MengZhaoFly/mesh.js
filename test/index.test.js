import fs from 'fs';
import path from 'path';

const jsCode = fs.readFileSync(
  path.resolve(__dirname, '../dist/mesh.js'),
  'utf-8',
);

describe('meshjs', () => {
  beforeAll(async () => {
    await page.setContent(`
      <canvas></canvas>
      <script>${jsCode}</script>
      <script>
        const {Renderer, Figure2D, Mesh2D} = meshjs
        const canvas = document.querySelector('canvas');
      </script>
    `);
  });

  it('render', async () => {
    const result = await page.evaluate(() => {
      const renderer = new Renderer(canvas);
      const figure = new Figure2D();
      figure.rect(50, 50, 100, 100);
      const mesh = new Mesh2D(figure, canvas);
      mesh.setFill({
        color: [1, 0, 0, 1],
      });
      renderer.drawMeshes([mesh]);
      return canvas.toDataURL();
    });
    expect(result).toMatchSnapshot();
  });
});
