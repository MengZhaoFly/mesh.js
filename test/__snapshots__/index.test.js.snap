// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meshjs render 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAACWCAYAAABkW7XSAAAGPUlEQVR4Xu3UgZGkQBDEwD3/jf73AkpN4gDV0oT+fj4EEEAgQuAvstNMBBBA4CdYHgECCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIICBY3gACCGQICFZGlaEIIHAmWP9+v3907hD4+/3OvK0dqpaceVSCtfWYBWvLx5U1gnXF5NgdgjUm5MgcwToicu0MwVozcmOPYN3wOHeFYM0pOTFIsE5o3DtCsPacXFgkWBcsDt4gWINSDkwSrAMSF08QrEUr/U2C1Xc4eYFgTWrJjxKsvMLNAwRr00t9lWDVDY7uF6xRMfFZghUXuDpfsFbNtHcJVtvf7HrBmlWTHiZYaX274wVr1015mWCV7Q1vF6xhOeFpghWWtzxdsJbtdLcJVtfd9HLBmtaTHSdYWXXbwwVr2091nWBVzY3vFqxxQdF5ghUVtz5bsNYNNfcJVtPb/GrBmleUHChYSW37owVr31FxoWAVrQU2C1ZAUnCiYAWlFSYLVsFSb6Ng9ZwlFgtWQlNupGDllDUGC1bDU22lYNWMRfYKVkRUbKZgxYRV5gpWxVRrp2C1fGXWClZGVWqoYKV0dcYKVsdVaalglWyFtgpWSFZoqmCFZJWmClbJVmerYHVcpZYKVkpXZqxgZVS1hgpWy1dlrWBVTMV2ClZMWGSuYEVE1WYKVs1YY69gNTzlVgpWTllisGAlNPVGClbPWWGxYBUsBTcKVlBaYLJgBSQVJwpW0dr+ZsHad5RcKFhJbfOjBWteUXOgYDW9ra8WrHVD0X2CFRU3PluwxgVV5wlW1dz2bsHa9pNdJ1hZddPDBWtaT3ecYHXdLS8XrGU74W2CFZY3PF2whuWUpwlW2d7udsHadZNeJlhpfbPjBWtWTXuYYLX9ra4XrFUz8V2CFRc4Ol+wRsXUZwlW3eDmfsHa9JJfJVh5hZMHCNaklv4oweo7XLxAsBatHNgkWAckDp4gWINSLkwSrAsW924QrD0nJxYJ1gmNc0cI1pySG4ME64bHtSsEa83IkT2CdUTk2BmCNSbkyhzBumJy6w7B2vJxZo1gnVE5dYhgTem4M0aw7rhcukSwlmwc2iJYh2QOnSJYQzIuTRGsSzZ3bhGsHRenlgjWKZ0zxwjWjIpbQwTrls+VawRrxcSxHYJ1TOjIOYI1IuLaDMG6ZnTjHsHa8HBuhWCdUzpxkGBNaLg3QrDuOV24SLAWLBzcIFgHpQ6cJFgDEi5OEKyLVt+/SbDed3BygWCd1Pr6UYL1uoKbAwTrpte3rxKstw0c/b9gHRX78lmC9bKAq78XrKtm371LsN7lf/bvgnVW7auHCdar+O/+XLDuun3zMsF6k/7hfwvWYbkvniZYL8K//GvBumz3vdsE6z32p/8sWKf1vnacYL2G/vaPBeu237euE6y3yB//r2AdF/zSeYL1EvjrvxWs64bfuU+w3uF+/q+CdV7xKwcK1ivY7/9UsO47fuNCwXqD+gf+KVgfkPzCiYL1AvQv/FKwvmD5+RsF63nmn/ijYH1C8+NHCtbjyL/xQ8H6huenrxSsp4l/5H+C9RHRD58pWA8D/8rvBOsrpp+9U7Ce5f2ZvwnWZ1Q/eqhgPYr7Oz8TrO+4fvJSwXqS9of+JVgfkv3gqYL1IOwv/UqwvmT7uVsF6znWn/qTYH1K92PHCtZjqL/1I8H6lu+nrhWsp0h/7D+C9THhD50rWA+B/tpvBOtrxp+59z/tT8iXpXy5MwAAAABJRU5ErkJggg=="`;
